Feature/296 upgrade scalatest from 3.0.5 to 3.2.15

AnilPinarOzdemir opened the pull request:
Scalatest updated from 3.0.5 to 3.2.15
Added new dependency "Scalatestplus" in order to use MockitoSugar.
TODO There is one error in TestClassLoaderUtils on Line 41.


Closes #296

AnilPinarOzdemir committed:
#296 Upgrade scalatest from 3.0.5 to 3.2.14

AnilPinarOzdemir committed:
#296 Upgrade scalatest from 3.0.5 to 3.2.14

Todo TestClassLoaderUtils.scala - test fails on line 41.

kevinwallimann commented on shared/src/test/scala/za/co/absa/hyperdrive/shared/utils/TestClassLoaderUtils.scala:
```diff
@@ -35,8 +35,10 @@ class TestClassLoaderUtils extends FlatSpec {
 
   // If this test fails, you might have to check TestClassLoaderUtilsOther.jar and createTestClassLoaderUtilsOtherJar.sh
-  it should "return the singleton class for a given trait and fully qualified class name from other jar" in {
+  ignore should "return the singleton class for a given trait and fully qualified class name from other jar" in {
```
It works on my machine. Does it always fail on your machine or is it flaky? Please try to un-ignore the test and see if the Github action would fail on this test

kevinwallimann reviewed (commented):
Thanks Anil for the PR, nicely done. It seems you forgot some tests for the Spark 3 / Scala 2.12 profile. You will see the test errors when you run `mvn clean test -Pscala-2.12,spark-3`

AnilPinarOzdemir committed:
#296 Upgrade scalatest from 3.0.5 to 3.2.14

Fixed test error for Spark-3 & Scala 2.12

TestClassLoaderUtils.scala Line 38 canged back to IT again, will test via pushing to GIT

AnilPinarOzdemir committed:
#296 Upgrade scalatest from 3.0.5 to 3.2.14

Fixed test error for Spark-3 & Scala 2.12

kevinwallimann commented:
Very nice! 🚀
